### DISCUSSION TITLE
[Feature request] 2D Histograms in seaborn.objects interface

### DISCUSSION THREAD

**Discussion by berkgercek** [mwaskom/seaborn#3200] [Q&A]
Heya,

Insofar as I'm aware, there is no way to use the existing objects in the new API to produce a 2D histogram. 

I'd really like this functionality, particularly because I'd like to be able to use something like the `seaborn.histplot` function from the old API, but with the option of adding annotations to each bin so I can see the stat associated with the bin in 2D space.

Currently to accomplish the above I'd have to manually bin the data using e.g. np.histogram2d and then pass it to `seaborn.heatmap` for plotting with `annot=True` kwarg. While this is totally doable, I see the opportunity for the really elegant compositionality of the objects interface to shine with this.

Thanks for your time!

**Comment by mwaskom** [comment]
Yes, expect support for this in the future.

**Comment by beyarkay** [comment]
Just piping in to say that I'd also really love this feature. Using a scatter plot with `alpha=0.2` isn't quite the same.
